sql/core/v1.2.1/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala

```diff
@@ -75,10 +75,42 @@ private[sql] object OrcFilters extends OrcFiltersBase {
       schema: StructType,
       dataTypeMap: Map[String, DataType],
       filters: Seq[Filter]): Seq[Filter] = {
-    for {
-      filter <- filters
-      _ <- buildSearchArgument(dataTypeMap, filter, newBuilder())
-    } yield filter
+    import org.apache.spark.sql.sources._
+
+    def convertibleFiltersHelper(
```

Comment:
it's weird to call a method "helper", what does it do?

Reply:
This is a helper method for converting Filter to Expression recursively.  It can also be `_convertibleFilters` or `convertibleFilters0`...
Here it is following `createFilterHelper` in `ParquetFilters`.